implementations/c/include/ockam/transport.h

```diff
@@ -1,147 +1,168 @@
+  kTestFailure                      = 0x0109, /*!< Error in test program */
+  kMalloc                           = 0x010a, /*!< Malloc failed */
+  kBadParameter                     = 0x010b  /*!< Bad parameter */
+} TransportError;
 
 /*
  ********************************************************************************************************
- *                                        PUBLIC DATA TYPES                                             *
+ *                                        PUBLIC DATA TYPES *
  ********************************************************************************************************
  */
 
-typedef enum {
-    CONNECTION_TYPE_UNDEFINED               = 0,
-    CONNECTION_TYPE_POSIX_TCP               = 1,
-    CONNECTION_TYPE_POSIX_UDP               = 2
-} OCKAM_CONNECTION_TYPE;
-
 /**
- * OCKAM_TRANSPORT_HANDLE represents a communication channel between two entities. Virtually every
- * call into the transport library takes an OCKAM_TRANSPORT_HANDLE as an argument.
+ * OckamInternetAddress - User-friendly internet addresses, includes
+ * terminating NULL
  */
-typedef void*           OCKAM_TRANSPORT_CONNECTION;
+typedef struct {
+  char dnsName[MAX_DNS_NAME_LENGTH];       // "www.name.ext"
+  char IPAddress[MAX_DNS_ADDRESS_LENGTH];  //"xxx.xxx.xxx.xxx"
+  uint16_t port;
+} OckamInternetAddress;
 
 /**
- * OCKAM_INTERNET_ADDRESS - User-friendly internet addresses, includes terminating NULL
+ * One Transmission instance is assigned for each read or write
  */
 typedef struct {
-    char					dns_name[MAX_DNS_NAME_LENGTH];			// "www.name.ext"
-    char					ip_address[MAX_DNS_ADDRESS_LENGTH]; 	//"xxx.xxx.xxx.xxx"
-    uint16_t                port;
-} OCKAM_INTERNET_ADDRESS;
+  void* buffer;                     // buffer to transmit (user-allocated)
+  uint16_t bufferSize;              // number of bytes to transmit (write) or buffer size (read)
+  uint16_t bytesTransmitted;        // number of bytes transmitted (so far)
+  TransportError completionStatus;  // transmission completion status
+} Transmission;
 
 /**
- * OCKAM_LISTEN_ADDRESS - Address on which to listen for an incoming connection request.
- * Address type will vary depending on connection type.
+ * The PosixSocket is the posix socket specific class data for a posix socket
+ * connection (TCP or UDP). Note that TCP sockets are further defined by the
+ * POSIX_TCP_SOCKET type.
  */
-typedef union {
-    OCKAM_INTERNET_ADDRESS              internet_address;
-} OCKAM_LISTEN_ADDRESS;
+typedef struct {
+  uint16_t isConnected;                // connection with remote is established
+  OckamInternetAddress localAddress;   // human-friendly local address
+  OckamInternetAddress remoteAddress;  // human-friendly remote address
+  int socket;                           // posix socket identifier
+  struct sockaddr_in socketAddress;    // network-friendly socket information
+  Transmission receiveTransmission;
+  Transmission sendTransmission;
+} PosixSocket;
```

Comment:
move all details related to Posix and TCP sockets to that impl header.

Reply:
done